Path: gag.com!mail-gateway
From: rhirst@linuxcare.com (Richard Hirst)
Subject: Re: [Linux-ia64] framebuffer console
Date: 4 Nov 2001 15:41:52 -0700
Organization: GAG News/Mail Gateway
Lines: 37
Message-ID: <20011104220637.B8938@linuxcare.com>
References: <20011024113033.M11105@linuxcare.com>
NNTP-Posting-Host: winfree.gag.com
X-Trace: winfree.gag.com 1004913712 25870 192.133.104.8 (4 Nov 2001 22:41:52 GMT)
X-Complaints-To: usenet@gag.com
NNTP-Posting-Date: 4 Nov 2001 22:41:52 GMT
Xref: gag.com list.linux.ia64:1284
Newsgroups: list.linux.ia64

On Wed, Oct 24, 2001 at 04:21:08PM -0700, David Mosberger wrote:
> >>>>> On Wed, 24 Oct 2001 12:26:59 +0100, Richard Hirst <rhirst@linuxcare.com> said:
> 
>   Richard> This patch solves the problem.  Still surprised that cached
>   Richard> accesses to the video memory crash the machine though..
> 
> My understanding is that Merced doesn't handle cache line sized
> transactions to the PCI bus.  Instead of going uncached, you might be
> able to use write-combining (for the video framebuffer, not for memory
> mapped control registers).  That can give you noticably better
> performance.

Yes, write combining works.  Could you include this in your tree
please, (patch is from a 2.4.9 tree)

Thanks,
  Richard



--- linux/drivers/video/fbmem.c.ori	Mon Nov  5 01:53:01 2001
+++ linux/drivers/video/fbmem.c	Mon Nov  5 01:53:05 2001
@@ -605,6 +605,8 @@
 	vma->vm_flags |= VM_IO;
 #elif defined(__sh__)
 	pgprot_val(vma->vm_page_prot) &= ~_PAGE_CACHABLE;
+#elif defined(__ia64__)
+	vma->vm_page_prot = pgprot_writecombine(vma->vm_page_prot);
 #else
 #warning What do we have to do here??
 #endif

_______________________________________________
Linux-IA64 mailing list
Linux-IA64@linuxia64.org
http://lists.linuxia64.org/lists/listinfo/linux-ia64

